src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -150,146 +147,79 @@ private static void ScavengeCache()
 					}
-					else if (indexOfUsingColon > 0 && originalPrefix[indexOfUsingColon - 1] == '#')
+
+					return methodName == nameof(ApiInformation.IsApiContractPresent) ?
+						ApiInformation.IsApiContractPresent(elements[1], majorVersion) :
+						ApiInformation.IsApiContractNotPresent(elements[1], majorVersion);
+				case nameof(ApiInformation.IsTypePresent):
+				case nameof(ApiInformation.IsTypeNotPresent):
+					if (elements.Length < 2)
 					{
-						// We have "#using:", we want to keep it.
-						adjusted
-							.Replace(
-								"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, originalPrefix),
-								"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, originalPrefix.Substring(indexOfUsingColon - 1))
-							);
+						throw new InvalidOperationException($"Syntax error while parsing conditional namespace expression {namespaceUri}");
 					}
-				}
+					var expectedType = elements[1];
+					return methodName == nameof(ApiInformation.IsTypePresent) ?
+						ApiInformation.IsTypePresent(elements[1], _metadataHelper) :
+						ApiInformation.IsTypeNotPresent(elements[1], _metadataHelper);
+				default:
+					return null;// TODO: support IsPropertyPresent
 			}
+		}
 
-			foreach (var includedCond in conditionals.IncludedConditionals)
-			{
-				var valueSplit = includedCond.Value.Split('?');
-				// Strip the conditional part, so the namespace can be parsed correctly by the Xaml reader
-				adjusted
-					.Replace(
-						includedCond.OuterXml,
-						"{0}=\"{1}\"".InvariantCultureFormat(includedCond.Name, valueSplit[0])
-					);
-			}
+		private XmlReader ApplyIgnorables(SourceText sourceText)
+		{
+			var originalString = sourceText.ToString();
 
-			if (hasxBind)
+			var document = new XmlDocument();
+			document.LoadXml(originalString);
+
+			//var conditionals = FindConditionals(document); /*TODO*/
+
+			var hasxBind = originalString.Contains("{x:Bind", StringComparison.Ordinal);
+
+			if (!hasxBind)
 			{
-				// Apply replacements to avoid having issues with the XAML parser which does not
-				// support quotes in positional markup extensions parameters.
-				// Note that the UWP preprocessor does not need to apply those replacements as the
-				// x:Bind expressions are being removed during the first phase and replaced by "connections".
-				adjusted = new(XBindExpressionParser.RewriteDocumentPaths(adjusted.ToString()));
+				// No need to modify file
+				return XmlReader.Create(new StringReader(originalString));
 			}
 
-			return (XmlReader.Create(new StringReader(adjusted.ToString())), conditionals.DisableCaching);
+			// Apply replacements to avoid having issues with the XAML parser which does not
+			// support quotes in positional markup extensions parameters.
+			// Note that the UWP preprocessor does not need to apply those replacements as the
+			// x:Bind expressions are being removed during the first phase and replaced by "connections".
+			var adjusted = XBindExpressionParser.RewriteDocumentPaths(originalString);
```

Comment:
This is the remaining costly string allocation. I'll see if I can do anything here, but that needs a separate PR.

It's still quite hard to get this PR green and will likely take some more time.

Reply:
Nothing is immediately clear on what would be an ideal fix here.

@jeromelaban Should I expect that we can get rid of this when we move away from mono parser?